### PR TITLE
Add back in stock form

### PR DESCRIPTION
**Pull request by dhmacs**
This is the first implementation of #688. When the new back in stock api is ready we'll make another PR to use that.

## QA

1. Open Vercel Preview
2. Visit a product without stock
3. Verify that you see the back in stock form 
image
4. Insert your email and verify that the request succeeds
5. Disable the network to simulate a network error, and verify that an error message is shown

### CODE REVIEW

**Comment by jordycosta** [comment]
QA 🟢 

Back-in-stock form is present for out-of-stock products 👍 
Request succeeds with network on, and fails when not connected to a network ✔️ 

---

### Testing Procedure:
1. Find an out-of-stock product via [inventory_report](https://me.cominor.com/Admin/Inventory/inventory_report.php)
2. Search this product in the Vercel [preview](https://react-commerce-git-add-back-in-stock-form-ifixit.vercel.app/)
3. Type in an email, click **Notify me** (this request should work)
4. Repeat step 3.) without being connected to a network

  Products tested:
  [ScrewMat Collection](https://react-commerce-git-add-back-in-stock-form-ifixit.vercel.app/Products/screwmat-collection) (Tool)
  [iPhone 6s Plus OEM Rear Case](https://react-commerce-git-add-back-in-stock-form-ifixit.vercel.app/Products/iphone-6s-plus-oem-rear-case) (Part)

---

### Screen Captures:

firefox_6iaP83UE2v

firefox_930LcEmOIR

File: packages/ui/theme/components/alert.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { ComponentStyleConfig } from '@chakra-ui/react';
+
+const Alert: ComponentStyleConfig = {
+   baseStyle: {},
+   variants: {
+      subtle: (props) => {
+         console.log('hey subtle');
```
Comment (masonmcelvain) [inline packages/ui/theme/components/alert.ts:7]: log

File: frontend/templates/product/sections/ProductSection/AddToCart/index.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+type InvetoryMessageProps = {
+   quantityAvailable?: number;
+};
+
+function InventoryMessage({ quantityAvailable }: InvetoryMessageProps) {
+   const theme = useTheme();
+   if (quantityAvailable == null || quantityAvailable >= 10) {
+      return null;
+   }
```
Comment (masonmcelvain) [inline frontend/templates/product/sections/ProductSection/AddToCart/index.tsx:122]: This is a duplicate conditional of the check we do on lines 61-62. We should drop one of them so they don't get out of sync.

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Nice!

File: frontend/templates/product/sections/ProductSection/AddToCart/index.tsx (hunk 2 above)
Comment (dhmacs) [reply frontend/templates/product/sections/ProductSection/AddToCart/index.tsx:122]: That's right 👍 I forgot to remove it

**Pull request merged**